src/input/type_record.cc

```diff
@@ -663,6 +729,68 @@ bool AbstractRecord::have_default_descendant() const {
 
 
 
+boost::shared_ptr<TypeBase> AbstractRecord::make_instance(std::vector<ParameterPair> vec) const {
+	AbstractRecord abstract = AbstractRecord(this->type_name(), this->child_data_->description_);
+	// Set default descendant
+	if (this->have_default_descendant()) {
+		abstract.allow_auto_conversion(child_data_->selection_default_.value());
+	}
+	// Copy attributes
+	abstract.attributes_ = boost::make_shared<attribute_map>(*attributes_);
+	// Set parameters as attribute
+	std::stringstream ss;
+	ss << "[";
+	for (std::vector<ParameterPair>::iterator vec_it=vec.begin(); vec_it!=vec.end(); vec_it++) {
+		if (vec_it != vec.begin()) ss << "," << endl;
+		ss << "{ \"" << (*vec_it).first << "\" : \"" << (*vec_it).second->content_hash() << "\" }";
+	}
+	ss << "]";
+	abstract.add_attribute("parameters", ss.str());
+	// Close abstract
+	abstract.close();
+
+	// make instances of all descendant records and add them into instance of abstract
+	for (ChildDataIter child_it = begin_child_data(); child_it != end_child_data(); ++child_it) {
+		Record rec = Record((*child_it).type_name(), (*child_it).data_->description_);
+		// Add parent Abstracts, generic (this) record is replaced by created instance
+		TypeHash hash = this->content_hash();
+		for (auto it = (*child_it).data_->parent_ptr_.begin(); it != (*child_it).data_->parent_ptr_.end(); ++it) {
+			if ( (*it)->content_hash() == hash ) {
+				rec.derive_from( abstract );
```

Comment:
Possible solution, Record can forgot its parent after registration, then Abstract has to call private method Record::add_parent() during its Abstract::finish()